src/concord-once.c

```diff
@@ -9,6 +9,12 @@ int ccord_has_sigint = 0;
 
 static int once;
 
+void
+ccord_shutdown_async(void)
+{
+    ccord_has_sigint = 1;
```

Comment:
Nit: because this function may be called from separate threads we should probably enclose the global var with mutexes, we can initialize a global lock for it like so:
```c
static pthread_mutex_t g_lock = PTHREAD_MUTEX_INITIALIZER;
```
And then use it like so:
```suggestion
    pthread_mutex_lock(&g_lock);
    ccord_should_shutdown = 1; // makes sense to rename this too
    pthread_mutex_unlock(&g_lock);
```
Thoughts?

Reply:
I guess that makes sense, although I'm not following how calling ccord_shutdown_async might be multithread unsafe on its own. Once the flag is set to 1, then any other calls to the function are basically a no op, right? i.e. it can only go from 0 to 1, or am I wrong about that?

Reply:
The flag is being checked at `discord_run()` so there should be a very slim (although not impossible) chance that it will read at the same time as `ccord_shutdown_async()` is writing to it.

See [here](https://github.com/Cogmasters/concord/blob/38dc9b2ea40cf4c87d81f7dd92639987fc8e946a/src/discord-loop.c#L114), we would also need to add locking here

But in all honesty, this is just me being pedantic

Reply:
I think this is good, but, I will say this, in the context of a complex program with many running threads all doing something:

From the library standpoint, it's going out and is going to be shut down, so, you might as well just terminate it; locking it truly wouldn't make a big difference.

From the program standpoint, it somewhat makes sense to lock the shutdown flag -- you might want to reactivate Concord without a total restart!

Reply:
Ah, I was anticipating ccord_shutdown_async would only be used on shutdowns. If not, then maybe some locking is needed. (In the case of a shutdown, I don't think it matters if there's a simultaneous read: `discord_read` might read 0 on one iteration, but eventually it's guaranteed to read 1, just perhaps on the next loop instead of the current one, so there's no wrong behavior if it's an inconsistent read.)

If we need to lock in discord-loop, a static lock won't work; should this be non-static and simply referenced extern in discord-loop.c, just like ccord_has_sigint?

If we're renaming things anyways, then maybe `shutdown_lock` is a more descriptive name than `g_lock`?

Reply:
> If we need to lock in discord-loop, a static lock won't work; should this be non-static and simply referenced extern in discord-loop.c, just like ccord_has_sigint?

Yup, or perhaps we wrap `ccord_has_sigint` in a function and then there's no need to be extern'ing variables? Ex
```c
int
discord_will_shutdown(void) {
  int retval;
  pthread_mutex_lock(&shutdown_lock);
  retval = ccord_will_shutdown;
  pthread_mutex_unlock(&shutdown_unlock);
  return retval;
}
```

> If we're renaming things anyways, then maybe shutdown_lock is a more descriptive name than g_lock?

Sure, I'm fine with it!

Reply:
Okay, I added another commit to the PR that hopefully addresses all these comments.